controllers/nmstate_controller.go

```diff
@@ -83,6 +82,13 @@ func (r *NMStateReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
+	// We only want one instance of NMState. Ignore anything after that.
+	if len(instanceList.Items) > 0 && instanceList.Items[0].Name != req.Name {
+		r.Log.Info("Ignoring NMState.nmstate.io because one already exists and does not match existing name")
+		err = r.Client.Delete(context.TODO(), instance, &client.DeleteOptions{})
+		return ctrl.Result{}, nil
+	}
+
```

Comment:
We can also put some code on the webhook so incase webhooks are installed it can be enforced at webhook level so no CRD is created.

Reply:
That's also a good idea. You want that in this PR? Or a separate one?

Reply:
Different PR